recipe/build.sh

```diff
@@ -24,7 +24,7 @@ fi
    --enable-special \
    ${OPENFST_CROSS_COMPILATION_CONFIGURE_OPTS}
 
-if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" && "$SHLIB_EXT" != '.dylib' ]]; then
```

Comment:
Isn't this more hiding the problem rather than fixing it?

Reply:
It is, but I don't see anything that the test executables are doing differently to link to libfst than the binaries, and those are all passing.  Rather, the tests aren't failing because something's been compiled wrong overall, it's specific to that one executable.  My next  idea at fixing it would be to have osx use CMake, since I understand it a lot better than the current autotools build.  Do you think it's worth that route when all the other functionality tests are passing?

Reply:
That's fair enough for the current approach, but I think there'd still be a lot of value in moving to CMake (ideally upstream in openfst directly)